### ISSUE TITLE
Fine-tuning with CLIP and accuracy decreases?

### ISSUE THREAD

**Issue by oh-livia** [openai/CLIP#161]
Has anyone tried to use CLIP fine-tuning as part of a larger model? I am getting some strange issues with accuracy jumping up to around 20% and then... dropping.  I cannot tell if this is just an issue with my model setup, or something I'm doing with the CLIP fine-tuning. The model uses the CLIP image embeddings for both images, concatenates them, then passes them through a few FC layers to get an output the same size as the CLIP text embedding. I use a CrossEntropy loss on the cosine measurements between the output and existing caption embeddings I have (and follow the advice from other posts about changing the weight types for CLIP finetuning).

```
        convert_models_to_fp32(clip_model)
        clip_opt.step()
        clip.model.convert_weights(clip_model)
```

I don't know how much detail is needed to showcase the issue I'm having, but I'm curious if anyone else has run into issues of accuracy decreasing when trying to fine-tune the CLIP embeddings

**Comment by dongyunkim-arch** [comment]
HI! I am also trying to do CLIP fine-tuning. I am wondering how you get the accuracy metrics since when I trained it, I could only see the loss, but it seems many people could also plot accuracy...

**Comment by Liaoqing-up** [comment]
hello, i also want to train the CLIP. How you train it ?

**Comment by Viditagarwal7479** [comment]
> HI! I am also trying to do CLIP fine-tuning. I am wondering how you get the accuracy metrics since when I trained it, I could only see the loss, but it seems many people could also plot accuracy...

@dongyun-kim-arch by evaluating it's zero-shot classification accuracy.

**Comment by Viditagarwal7479** [comment]
> Has anyone tried to use CLIP fine-tuning as part of a larger model? I am getting some strange issues with accuracy jumping up to around 20% and then... dropping. I cannot tell if this is just an issue with my model setup, or something I'm doing with the CLIP fine-tuning. The model uses the CLIP image embeddings for both images, concatenates them, then passes them through a few FC layers to get an output the same size as the CLIP text embedding. I use a CrossEntropy loss on the cosine measurements between the output and existing caption embeddings I have (and follow the advice from other posts about changing the weight types for CLIP finetuning).
> 
> ```
>         convert_models_to_fp32(clip_model)
>         clip_opt.step()
>         clip.model.convert_weights(clip_model)
> ```
> 
> I don't know how much detail is needed to showcase the issue I'm having, but I'm curious if anyone else has run into issues of accuracy decreasing when trying to fine-tune the CLIP embeddings

@oh-livia I am also facing similar issue, the accuracy keeps on dropping even after trying various learning rates and gradient accumulation steps. Did you find any solution for this?
